### ISSUE TITLE
Renovate unable to resolve private shareable preset

### ISSUE THREAD

**Issue by hisabimbola** [renovatebot/renovate#9665]
**What Renovate type, platform and version are you using?**

Self-hosted Renovate
Version: renovate/renovate:24.119.4

**Describe the bug**

When a preset is stored in a private repo, renovate is not able to fetch the preset.

**Relevant debug logs**
I created a sample repo [here](https://github.com/hisabimbola/github-renovate/runs/2398774708?check_suite_focus=true)

You can see that it could not fetch the preset repo, even though there is a valid token.

<details><summary>Click me to see logs</summary>

```
DEBUG: Using RE2 as regex engine
DEBUG: Parsing configs
DEBUG: Checking for config file in /github/workspace/dist/src/renovate-config.js
DEBUG: GitHub 404
       "url": "https://api.github.com/repos/hisabimbola/renovate-base/contents/base.json"
DEBUG: Failed to retrieve base.json from repo
       "statusCode": 404,
       "url": "https://api.github.com/repos/hisabimbola/renovate-base/contents/base.json"
DEBUG: Preset fetch error
       "preset": "github>hisabimbola/renovate-base:base",
       "err": {
         "message": "dep not found",
         "stack": "Error: dep not found\n    at fetchJSONFile (/usr/src/app/node_modules/renovate/lib/config/presets/github/index.ts:33:11)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at Object.fetchPreset (/usr/src/app/node_modules/renovate/lib/config/presets/util.ts:59:19)\n    at getPreset (/usr/src/app/node_modules/renovate/lib/config/presets/index.ts:184:22)\n    at Object.resolveConfigPresets (/usr/src/app/node_modules/renovate/lib/config/presets/index.ts:259:27)\n    at Object.parseConfigs (/usr/src/app/node_modules/renovate/lib/config/index.ts:56:22)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:76:14)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
       }
 INFO: Throwing preset error
       "validationError": "Cannot find preset's package (github>hisabimbola/renovate-base:base)"
FATAL: Fatal error: config-validation
       "err": {
         "validationError": "Cannot find preset's package (github>hisabimbola/renovate-base:base)",
         "message": "config-validation",
         "stack": "Error: config-validation\n    at Object.resolveConfigPresets (/usr/src/app/node_modules/renovate/lib/config/presets/index.ts:270:25)\n    at processTicksAndRejections (internal/process/task_queues.js:93:5)\n    at Object.parseConfigs (/usr/src/app/node_modules/renovate/lib/config/index.ts:56:22)\n    at Object.start (/usr/src/app/node_modules/renovate/lib/workers/global/index.ts:76:14)\n    at /usr/src/app/node_modules/renovate/lib/renovate.ts:16:22"
       }
DEBUG: Missing config
DEBUG: Renovate exiting
```

</details>

**Have you created a minimal reproduction repository?**

Please read the [minimal reproductions documentation](https://github.com/renovatebot/renovate/blob/master/docs/development/minimal-reproductions.md) to learn how to make a good minimal reproduction repository.

- [x] I have provided a minimal reproduction repository
- [ ] I don't have time for that, but it happens in a public repository I have linked to
- [ ] I don't have time for that, and cannot share my private repository
- [ ] The nature of this bug means it's impossible to reproduce publicly

**Additional context**

Initial discussion about this issue [here](https://github.com/renovatebot/renovate/discussions/9638)

**Comment by hisabimbola** [comment]
Issue is fixed in v25 release #9394

**Comment by rarkins** [comment]
Great, thanks for your patience getting to the root of it
